hack/tools/tilt-prepare/main.go

```diff
@@ -470,16 +515,53 @@ func writeIfChanged(prefix string, path string, yaml []byte) error {
 	return nil
 }
 
-// prepareDeploymentForDebug alter controller deployments for working nicely with delve debugger;
-// most specifically, liveness and readiness probes are dropper and leader election turned off.
-func prepareDeploymentForDebug(prefix string, objs []unstructured.Unstructured) error {
+// prepareDeploymentCommandAndArgs sets the default Command and Args for the manager container.
+// extraArgs comes from the extra_args field in the tilt-settings file and are appended to the pre-existing ones.
+// This has the affect that if there are args already set in the manager Deployment the given ones via extra_args will
+// take precedence.
+func prepareDeploymentCommandAndArgs(prefix string, objs []unstructured.Unstructured, extraArgs extraArgs) error {
 	return updateDeployment(prefix, objs, func(d *appsv1.Deployment) {
 		for j, container := range d.Spec.Template.Spec.Containers {
 			if container.Name != "manager" {
 				// as defined in clusterctl Provider Contract "Controllers & Watching namespace"
 				continue
 			}
 
+			container.Command = []string{"sh", "/start.sh", "/manager"}
+			container.Args = append(container.Args, []string(extraArgs)...)
+
+			d.Spec.Template.Spec.Containers[j] = container
+		}
+	})
+}
+
+// prepareDeploymentForDebug alters controller deployments for working nicely with delve debugger;
+// most specifically, configuring delve, starting the manager with profiling enabled, dropping liveness and readiness
+// probes and disabling leader election.
+func prepareDeploymentForDebug(prefix string, objs []unstructured.Unstructured, debugConfig *debugConfig) error {
+	return updateDeployment(prefix, objs, func(d *appsv1.Deployment) {
+		for j, container := range d.Spec.Template.Spec.Containers {
+			if container.Name != "manager" {
+				// as defined in clusterctl Provider Contract "Controllers & Watching namespace"
+				continue
+			}
+
+			cmd := []string{"sh", "/start.sh", "/dlv", "--accept-multiclient", "--api-version=2", "--headless=true", "exec"}
+
+			if debugConfig.Port > 0 {
+				cmd = append(cmd, "--listen=:30000")
```

Comment:
Should me make this actually configurable? Same for profile and metrics port.

Reply:
I think we don't have to. Making it configurable would just make it more complex. We only have to live with the dependency between the Tiltfile and tilt-prepare. But I would prefer that.